src/lib/importMovieList.js

```diff
@@ -10,20 +10,36 @@ const options = {
   }
 };
 
-function transformUrl(url) {
+async function resolveShortenedUrl(url) {
+  const response = await fetch(url, { method: 'HEAD', redirect: 'follow' });
+  return response.url;
+}
+
+export async function transformUrl(url) {
+  // Example url: https://letterboxd.com/bluevoid/list/top-100-2020-edition/
   try {
+    if (url.includes('boxd.it')) {
+      const resolvedUrl = await resolveShortenedUrl(url);
+      if (!resolvedUrl.includes('letterboxd.com')) {
+        throw new Error('Invalid URL');
+      }
+      return transformUrl(resolvedUrl);
+    }
     const urlParts = url.split('/');
-    const username = urlParts[urlParts.length - 4];
-    const listName = urlParts[urlParts.length - 2];
+    if (urlParts.length < 5) {
```

Comment:
```suggestion
    if (urlParts.length < 6) {
```
Looks like the 6th element is relevant `const listName = urlParts[5];`

Reply:
Ah, yeah good catch.